### PR TITLE
Migrate access control example to documentation website

### PR DESCRIPTION
**Pull request by ADubhlaoich**
### Proposed changes

This pull request migrates the access control example to the documentation website, converting the existing README into a page formatted for the production website. It also cleans up the frontmatter of the surrounding documents so that pages can be more easily re-arranged in the future.

The converted page makes use of single source embedded code blocks using files in the GitHub repository, as well as a highlighting feature of code blocks that has not been used in previous documentation.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: examples/custom-resources/access-control/README.md [hunk 1]
```diff
@@ -1,96 +1,3 @@
-# Access Control
+# Deploy an Access Control policy
```
Comment (vepatel) [inline examples/custom-resources/access-control/README.md:1]: Access Control is a feature and policy is one of the steps so the original title is apt

Comment (ADubhlaoich) [reply examples/custom-resources/access-control/README.md:1]: The original title does not describe the intent of the document, and the new one is matched to the website document, which is written as an instruction.

File: site/content/configuration/access-control.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+For ease of use in shell commands, set two shell variables:
+
+1. The public IP address for your NGINX Ingress Controller instance.
+
+```shell
+IC_IP=<ip-address>
+```
+
+2. The HTTP port of the same instance
```
Comment (jputrino) [inline site/content/configuration/access-control.md:26]:
```suggestion
2. The HTTP port of the same instance.
```

File: site/content/configuration/access-control.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+---
+
+## Deploy the example application
+
+Create the file *webapp.yaml* with the following contents:
+
+{{< ghcode "https://raw.githubusercontent.com/nginxinc/kubernetes-ingress/refs/heads/main/examples/custom-resources/access-control/webapp.yaml" >}}
+
+Apply it using *kubectl*:
```
Comment (jputrino) [inline site/content/configuration/access-control.md:40]: Should 'kubectl' be formatted as code instead?

File: site/content/configuration/access-control.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+```shell
+kubectl apply -f access-control-policy-deny.yaml
+```
+
+---
+
+## Configure load balancing
+
+Create a file named *virtual-server.yaml* for the VirtualServer resource. The *policies* field references the example application.
```
Comment (jputrino) [inline site/content/configuration/access-control.md:64]:
```suggestion
Create a file named *virtual-server.yaml* for the VirtualServer resource. The *policies* field references the access control Policy created in the previous section.
```

File: site/content/configuration/access-control.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+Apply it using *kubectl*:
+
+```shell
+kubectl apply -f webapp.yaml
+```
+
+---
+
+## Deploy an Access Control policy
```
Comment (jputrino) [inline site/content/configuration/access-control.md:48]:
```suggestion
## Deploy a Policy to deny application access
```

File: site/content/configuration/access-control.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+---
+
+## Deploy the example application
+
+Create the file *webapp.yaml* with the following contents:
+
+{{< ghcode "https://raw.githubusercontent.com/nginxinc/kubernetes-ingress/refs/heads/main/examples/custom-resources/access-control/webapp.yaml" >}}
+
+Apply it using *kubectl*:
```
Comment (ADubhlaoich) [inline site/content/configuration/access-control.md:40]:
```suggestion
Apply it using 'kubectl':
```

File: site/content/configuration/access-control.md (hunk 5 above)
Comment (ADubhlaoich) [reply site/content/configuration/access-control.md:48]: The focus of the document as a whole is to show how to deploy an Access Control policy: the deny and allow steps existence in sequence to demonstrate this, so I don't think it makes sense to make the policy the focus nor is it necessary to be this explicit.

File: site/content/configuration/access-control.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+<center><h1>403 Forbidden</h1></center>
+</body>
+</html>
+```
+
+The *403* response is expected, successfully blocking your machine.
+
+---
+
+## Update the policy
```
[the remark this answers is not in the document]
Comment (ADubhlaoich) [reply site/content/configuration/access-control.md:96]: @vepatel Within the context of Kubernetes configuration, I have always thought of a Policy as the live configuration being updated by YAML files.

Is it more accurate to describe the YAML files themselves as separate Policies? The YAML files are ephemeral, so it feels odd to me.

**Review by vepatel — APPROVED** [review]

File: site/content/configuration/access-control.md (hunk 5 above)
Comment (jputrino) [reply site/content/configuration/access-control.md:48]: You're using the same text as the title of the document in a sub-heading, which is confusing for users. If the broad goal of the doc is to deploy an access control policy, and I complete this section, am I now done with the doc? Why are there more sections? It makes more sense to me to have the sub-headings accurately reflect the expected outcome of completing the steps in the section.

File: site/content/configuration/access-control.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jputrino) [reply site/content/configuration/access-control.md:96]: We are not asking the user to apply an update to their policy here; we're asking them to apply a new policy file, with a different name. Just from that standpoint, the heading of "update the policy" is misleading. Aside from that, the subheading should accurately reflect what the user is being instructed to do. In this case, that is deploying a policy to allow traffic. 

This also helps with SEO, so people who are searching specifically for how to deploy an allow or deny policy will easily find the section with the right instructions.

[the remark this answers is not in the document]
Comment (pdabelf5) [reply site/content/configuration/access-control.md:96]: Just to clarify, the Policy here is called `webapp-policy` so even though the updates are in two separate files, only one Policy is being updated.

**Review by j1m-ryan — APPROVED** [review]

**Review by pdabelf5 — APPROVED** [review]

File: site/content/configuration/access-control.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
+---
+title: Deploy an Access Control policy
+weight: 900
+toc: true
+docs: DOCS-000
+---
+
+This topic describes how to apply and update an Access Control policy with F5 NGINX Ingress Controller.
```
[the remark this answers is not in the document]
Comment (ADubhlaoich) [reply site/content/configuration/access-control.md:8]: I'll think a bit about this - the challenge around it is that policy exists as a general noun, whereas a Policy is a resource in context. 

There's technically no such thing as an "Access Control Policy" in terms of specificity, so I might change this to something along the lines of "Managing access control using a Policy".

[the remark this answers is not in the document]
Comment (j1m-ryan) [reply site/content/configuration/access-control.md:8]: This is a nit tbh in this PR as theres instances both ways in the examples folder. But yeah its something to think about.

**Pull request merged**